File: TowerOfHanoi/tower.py
def tower(disks, startRing, tempRing, endRing):
    if disks == 1:
        print('Move disk 1 from tower {} to tower {}.'.format(startRing, endRing))
        return
 
    tower(disks - 1, startRing, endRing, tempRing)
    print('Move disk {} from tower {} to tower {}.'.format(disks, startRing, endRing))
    tower(disks - 1, tempRing, startRing, endRing)
 
 
disks = 3
tower(disks, 'A', 'B', 'C')

File: TowerOfHanoi/tower.cs
public class tower
    {
        public static void Main(String[] args)
        {
            char startRing = '1'; // start tower in output
            char endRing = '2'; // end tower in output
            char tempRing = '3'; // temporary tower in output
            int totalDisks = 3; // number of disks
 
            solveTowers(totalDisks, startRing, endRing, tempRing);
        }
 
        private static void solveTowers(int n, char startRing, char endRing, char tempRing)
        {
            if (n > 0)
            {
                solveTowers(n - 1, startRing, tempRing, endRing);
                Console.WriteLine("Move disk from " + startRing + ' ' + endRing);
                solveTowers(n - 1, tempRing, endRing, startRing);
 
            }
        }        
    }